apis/cloud.redhat.com/v1alpha1/clowdapp_types.go

```diff
@@ -203,6 +203,20 @@ type CyndiSpec struct {
 	InsightsOnly bool `json:"insightsOnly,omitempty"`
 }
 
+// KafkaTopicSpec described the desired configuration of kafka topics
```

Comment:
Is there a reason we are creating a new object here, when we could use the strimzi one. I realise that previously we were writing a spec and putting it in the `strimzi` package, but the idea was that it "closely" matched the REAL object so that we could ensure that we got the right yaml/json generated.

Reply:
Yeah, see the comment just below inside the type def. "topicName" is actually an optional attribute in the real KafkaTopicSpec (if topicName is under, strimzi just uses the metadata.name of the KafkaTopic). We want name to be required, so I'm redefining the type here basically to represent our "clowdapp topic config".

If we want to make partitions and replicas optional here, we could set that in this type as well. The real KafkaTopic sets those as required params, so Clowder would be in charge of defaulting them to `1`

Reply:
Hmm I see - I know it's more for us to look into - but if we have now gone to the effort to get a full KafkaTopic spec, we could use a hook to validate and reject if it isn't set. - Thoughts?

Reply:
Or- I mean - we can just validate it in the reconciler

Reply:
I just updated topics/partitions to be optional, we will default to 3 if not present.

So our ClowdApp has:
replicas optional
partitions optional
topicName required

Whereas the [strimzi spec](https://github.com/bsquizz/strimzi-client-go/blob/main/apis/kafka.strimzi.io/v1beta1/kafkatopic_types.go#L50-L58) has:
replicas required
partitions required
topicName optional

So that's why I have a separate type defined here -- to correctly define what is optional/required to set on a ClowdApp